src/scheduler/server_info.cpp

```diff
@@ -441,8 +441,8 @@ query_server(status *pol, int pbs_sd)
 	 * we don't want to account for resources consumed by ghost jobs
 	 */
 	create_placement_sets(policy, sinfo);
-	if (!sinfo->node_group_enable && sinfo->node_group_key != NULL &&
-	    strcmp(sinfo->node_group_key[0], "msvr_node_group") == 0) {
+	if (!sinfo->node_group_enable && !sinfo->node_group_key.empty() &&
+	    (sinfo->node_group_key[0] == std::string("msvr_node_group"))) {
```

Comment:
No need to convert msvr_node_group into a string.  operator= of string is overloaded to take a char *.